datacube/api/core.py

```diff
@@ -940,6 +942,13 @@ def load_data(sources: xarray.DataArray, geobox: GeoBox,
         .. seealso:: :meth:`find_datasets` :meth:`group_datasets`
         """
         measurements = per_band_load_data_settings(measurements, resampling=resampling, fuse_func=fuse_func)
+        if driver := extra.pop('driver', None):
```

Comment:
If we're adding `driver` as an explicit new argument to `load` above I can't see why we wouldn't add it as an explicit new argument to `load_data` here as well.